modules/core/arrow-typeclasses/src/main/kotlin/arrow/typeclasses/Monad.kt

```diff
@@ -48,15 +48,36 @@ interface Monad<F> : Selective<F> {
   fun <A, B> Kind<F, A>.followedByEval(fb: Eval<Kind<F, B>>): Kind<F, B> =
     flatMap { fb.value() }
 
+  @Deprecated(
+    "effectM is being renamed to flatTap",
+    ReplaceWith("flatTap")
+  )
   fun <A, B> Kind<F, A>.effectM(f: (A) -> Kind<F, B>): Kind<F, A> =
+    flatTap(f)
+
+  fun <A, B> Kind<F, A>.flatTap(f: (A) -> Kind<F, B>): Kind<F, A> =
```

Comment:
Where does `flatTap` come from?

Reply:
It seems they have moved to the this naming model in Cats [FlatMap](https://github.com/typelevel/cats/blob/f59a9b0d250c90bca85e6d3a5fdfcc076be453cb/core/src/main/scala/cats/FlatMap.scala#L150).

Reply:
I don't like it but I guess it'll pass >.<

Reply:
Can't find a similar signature on Hoogle. I remember seeing `flatTap` in other libs as well but I'm drawing a blank right now.

Reply:
the rationale here is that `effect` is associated to suspend so it should not be used in places where there is no suspension or it will become confusing to users